src/python/pants/engine/rules.py

```diff
@@ -281,7 +281,7 @@ def rule_decorator(func, **kwargs) -> Callable:
     if effective_desc is None and is_goal_cls:
         effective_desc = f"`{effective_name}` goal"
 
-    effective_level = kwargs.get("level", LogLevel.DEBUG)
+    effective_level = kwargs.get("level", LogLevel.TRACE)
```

Comment:
It's possible that a rule should still default to Debug, and that we override certain noisy rules to Trace. But, still, this feels trace-y to me.

Imagine if we logged every time you called a normal Python function.

Reply:
Yeah, this is almost the definition of "trace".